### DISCUSSION TITLE
Why do custom errors thrown by target contracts not appear on Etherscan?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [PaulRBerg/prb-proxy#36] [General]
@cleanunicorn asked me this question privately, and it's a very good question. PRBProxy itself does bubble up the custom error thrown by a target contract, as per the code on lines 101-106:

https://github.com/paulrberg/prb-proxy/blob/69321207fdcee2ad810e0c30dfd120e4221edf0a/contracts/PRBProxy.sol#L101-L106

The issue is that Etherscan does not decode the error - even if it could very well do it, provided the target contract has its source code verified.

I have contacted Etherscan multiple times about this, kindly asking them to add support for bubbled up custom errors. They said that can't find the time to add this to their roadmap.

**Comment by alphak3y** [comment]
Was your custom error actually included in the ABI of the main contract in question?

I wonder if including these custom errors will help.

If not, this would make me sad as I am using a lot of external libraries currently which have custom errors.

**Reply by PaulRBerg** [reply]
Yep. The custom error was part of the ABI.

[I have continued to prod Etherscan](https://twitter.com/paulrberg/status/1640258888992800768) about adding support for custom errors.

**Comment by alphak3y** [comment]
Great. I have also created a [short thread providing full detail](https://twitter.com/alphaK3Y/status/1650147404182859777).

Maybe if I deploy contracts which have a large number of users and they all complain about this we can get it through.

**Reply by PaulRBerg** [reply]
That's awesome, thanks for the bump

**Reply by alphak3y** [reply]
To get around this I'm going to try:

`if (globalState.unlocked == 2) require(false, 'Locked()');`

It uses more bytecode size but should be the same gas costs in the non-revert case.

**Reply by PaulRBerg** [reply]
This is unfortunately a self-defeating approach because it means resorting back to revert reason strings.

**Reply by alphak3y** [reply]
If Tenderly is able to decode these errors, I can deploy with custom errors no problem.

Just need a debugging tool to provide support in case a user has a failed transaction.

**Comment by SvenMeyer** [comment]
Is this still an issue ?
I created a small swap contract with custom errors, forced it to revert, but when I checked etherscan all I saw was `Fail` :-(
